### PR TITLE
Add script to generate embedding for dataset

### PR DESCRIPTION
**Pull request by pkooij**
This PR introduces a way to generate image and text embeddings to a dataset to be more efficient when training a dataset for multiple epochs. For example for learning a general reward we contain a specific dataset with OXE to improve generalization. In order to nor recompute the image and text embeddings each time we finetune for OXE we can use this script to add the embeddings to the dataset. We can additionally remove the videos in the dataset to safe space. 

Testing:
Both the generate and validate script were tested on this dataset: [lerobot/utokyo_xarm_bimanual](https://huggingface.co/datasets/lerobot/utokyo_xarm_bimanual/tree/main/data/chunk-000). The generated dataset can be found here: [pepijn223/utokyo_xarm_bimanual_embeddings](https://huggingface.co/datasets/pepijn223/utokyo_xarm_bimanual_embeddings).

### CODE REVIEW

File: src/lerobot/datasets/generating_embeddings/generate_embeddings.py [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+                    # Load from image file
+                    img = item[cam_key]
+                    # Convert to numpy if needed
+                    if isinstance(img, torch.Tensor):
+                        if img.ndim == 3:
+                            img_np = (img.permute(1, 2, 0).numpy() * 255).astype(np.uint8)
+                        else:
+                            raise ValueError(f"Unexpected image shape {img.shape} for camera {cam_key}")
+                    else:
+                        img_np = np.array(img)
```
Comment (michel-aractingi) [inline src/lerobot/datasets/generating_embeddings/generate_embeddings.py]: Should we add testing for this part, 
this logic could be buggy if we have image datasets instead of videos.
Better if we had tests to verify

File: src/lerobot/datasets/generating_embeddings/encoders.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    DinoV2 is a self-supervised vision transformer that produces high-quality image embeddings.
+    Supports multiple model sizes (ViT-S/14, ViT-B/14, ViT-L/14).
+    """
+
+    def __init__(self, model_name: str = "dinov2_vitb14", device: str = "cuda", batch_size: int = 32):
+        super().__init__(device)
+        self.batch_size = batch_size
+        self.model_name = model_name
+        logger.info(f"Loading DinoV2 model: {model_name}")
+        self.model = torch.hub.load("facebookresearch/dinov2", model_name)  # nosec B614
```
Comment (michel-aractingi) [inline src/lerobot/datasets/generating_embeddings/encoders.py:50]: Is there a reason why you didn't use AutoModel from transformers here also? We do it for the SAC encoder. https://github.com/huggingface/lerobot/blob/6f5bb4d4a49fbdb47acfeaa2c190b5fa125f645a/src/lerobot/policies/sac/modeling_sac.py#L941
Something like:
```
        self.model = AutoModel.from_pretrained("facebook/dinov2_base"):
```

File: src/lerobot/datasets/generating_embeddings/validate_embeddings.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+from lerobot.datasets.generating_embeddings.encoders import ImageEncoder, LanguageEncoder
+from lerobot.datasets.generating_embeddings.generate_embeddings import (
+    get_image_encoder,
+    get_language_encoder,
+)
+from lerobot.datasets.lerobot_dataset import LeRobotDataset
+
+
+def cosine_similarity(a: np.ndarray, b: np.ndarray) -> float:
```
Comment (michel-aractingi) [inline src/lerobot/datasets/generating_embeddings/validate_embeddings.py:43]: nice

File: src/lerobot/datasets/generating_embeddings/generate_embeddings.py [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+        # Add embedding columns
+        df["task_embedding"] = [all_task_embeddings[idx].tolist() for idx in frame_indices]
+
+        for cam_key in dataset.meta.camera_keys:
+            df[f"{cam_key}_embedding"] = [
+                image_embeddings_dict[cam_key][idx].tolist() for idx in frame_indices
+            ]
+
+        # Save the updated parquet file
+        df.to_parquet(parquet_path, index=False)
```
Comment (michel-aractingi) [inline src/lerobot/datasets/generating_embeddings/generate_embeddings.py]: For this chunk I suggest to do it in a more effiecient way either by using dataset_tools. I'll push a PR.

Reading a writing the parquet file to disk will be slow and result in memory explosion for huge datasets.
